testnet/stacks-node/src/tests/signer/v0.rs

```diff
@@ -2643,3 +2642,500 @@ fn signer_set_rollover() {
+            let mined_before_1 = blocks_mined1.load(Ordering::SeqCst);
+            let mined_before_2 = blocks_mined2.load(Ordering::SeqCst);
+            let proposed_before_2 = blocks_proposed2.load(Ordering::SeqCst);
+
+            // submit a tx so that the miner will mine an extra block
+            let sender_nonce = (btc_blocks_mined - 1) * inter_blocks_per_tenure + interim_block_ix;
+            let transfer_tx =
+                make_stacks_transfer(&sender_sk, sender_nonce, send_fee, &recipient, send_amt);
+            submit_tx(&http_origin, &transfer_tx);
+
+            wait_for(60, || {
+                let mined_1 = blocks_mined1.load(Ordering::SeqCst);
+                let mined_2 = blocks_mined2.load(Ordering::SeqCst);
+                let proposed_2 = blocks_proposed2.load(Ordering::SeqCst);
+
+                Ok((fork_initiated && proposed_2 > proposed_before_2)
+                    || mined_1 > mined_before_1
+                    || mined_2 > mined_before_2)
+            })
+            .unwrap();
+            info!(
+                "Attempted to mine interim block {}:{}",
+                btc_blocks_mined, interim_block_ix
+            );
+        }
+
+        if miner == 1 {
+            miner_1_tenures += 1;
+        } else {
+            miner_2_tenures += 1;
+        }
+        info!(
+            "Miner 1 tenures: {}, Miner 2 tenures: {}",
+            miner_1_tenures, miner_2_tenures
+        );
+
+        let mined_1 = blocks_mined1.load(Ordering::SeqCst);
+        let mined_2 = blocks_mined2.load(Ordering::SeqCst);
+        if miner == 1 {
+            assert_eq!(mined_1, mined_before_1 + inter_blocks_per_tenure + 1);
+        } else {
+            if miner_2_tenures < min_miner_2_tenures {
+                assert_eq!(mined_2, mined_before_2 + inter_blocks_per_tenure + 1);
+            } else if miner_2_tenures == min_miner_2_tenures {
+                // If this is the forking tenure, miner 2 should have mined 0 blocks
+                assert_eq!(mined_2, mined_before_2);
+
+                // Clear the ignore block
+                clear_ignore_block();
+            }
+        }
+    }
+
+    info!(
+        "New chain info 1: {:?}",
+        get_chain_info(&signer_test.running_nodes.conf)
+    );
+
+    info!("New chain info 2: {:?}", get_chain_info(&conf_node_2));
+
+    let peer_1_height = get_chain_info(&conf).stacks_tip_height;
+    let peer_2_height = get_chain_info(&conf_node_2).stacks_tip_height;
+    info!("Peer height information"; "peer_1" => peer_1_height, "peer_2" => peer_2_height, "pre_naka_height" => pre_nakamoto_peer_1_height);
+    assert_eq!(peer_2_height, ignore_block - 1);
+    assert_eq!(
+        peer_1_height,
+        pre_nakamoto_peer_1_height
+            + (miner_1_tenures + min_miner_2_tenures - 1) * (inter_blocks_per_tenure + 1)
+    );
+    assert_eq!(
+        btc_blocks_mined,
+        u64::try_from(miner_1_tenures + miner_2_tenures).unwrap()
+    );
```

Comment:
Can you add test coverage here to inspect the chainstate of miner 2 to verify that the block was indeed ignored (i.e. it's absent in the chainstate)?  Also, there should be no sibling blocks in miner 2 versus miner 1, since the signers should have prevented miner 2 from forking the chain back to the start of miner 1's tenure (since the time-out to do so would have already passed).

Reply:
Sure, I'll figure out a good way to do that tomorrow.

Reply:
Is this sufficient? 8a160dc5d5cfc0ead5da5a9f3c50e97562f7f0ac
Anything else you'd like to see?

Reply:
Ah, what I mean is, there should be two tenure-start blocks at height `ignore_block` in miner 1, but only one tenure-start block at height `ignore_block` in miner 2 since miner 2 ignored it.

Reply:
Sure, I'll add this. I've also found that there is still some inconsistency in this test, so I will fix that.